File: src/com/damnae/osukeysoundsplitter/pathprovider/CounterKeysoundPathProvider.java
package com.damnae.osukeysoundsplitter.pathprovider;

public class CounterKeysoundPathProvider extends BaseKeysoundPathProvider {
	private String keysoundsFolderName;
	private int fileIndex;

	public CounterKeysoundPathProvider() {
	}

	public CounterKeysoundPathProvider(String keysoundsFolderName) {
		this.keysoundsFolderName = keysoundsFolderName;
	}

	@Override
	protected String getNewKeysoundPath(String extension) {
		++fileIndex;

		if (keysoundsFolderName == null)
			return fileIndex + "." + extension;

		return keysoundsFolderName + "/" + fileIndex + "." + extension;
	}
}

File: osu!KeysoundSplitter/PathProvider/CounterKeysoundPathProvider.cs

namespace osuKeysoundSplitter.PathProvider
{
    public class CounterKeysoundPathProvider : BaseKeysoundPathProvider
    {
        private string keysoundsFolderName;
        private int fileIndex;

        public CounterKeysoundPathProvider(KeysoundCache keysoundCache)
            : base(keysoundCache)
        {
        }

        public CounterKeysoundPathProvider(KeysoundCache keysoundCache,
                string keysoundsFolderName)
            : base(keysoundCache)
        {
            this.keysoundsFolderName = keysoundsFolderName;
        }

        protected override string getNewKeysoundPath(string extension)
        {
            ++fileIndex;

            if (keysoundsFolderName == null)
                return fileIndex + "." + extension;

            return keysoundsFolderName + "/" + fileIndex + "." + extension;
        }
    }
}